Revision Questions - Java - Blockchain (#2830)
enkidevs/curriculum: java/java-fundamentals/exceptions-i/throwing-an-exception.md

category: must-know
links:
  - >-
    [How to Handle Exceptions Better](http://crunchify.com/better-understanding-on-checked-vs-unchecked-exceptions-how-to-handle-exception-better-way-in-java/){website}
notes: |
  https://insights.enki.com/#/contrib/55a3b45d2a4a89320042e651
revisionQuestion:
  formats:
    - fill-in-the-gap
    - type-in-the-gap
  context: standalone
---

# Throwing an `exception`

---

## Content

> An `exception` is an event that occurs during the execution of a program disrupting the normal flow of the program.

Reviewer: We should change `exception` to _exception_ in this whole insight because "exception" is not a keyword

Reviewer: ```suggestion
An *exception* is an event that occurs during the execution of a program disrupting the normal flow of the program.
```

> There are two types of `exception`:

Reviewer: ```suggestion
There are two types of *exceptions*:
```

> - *Checked* : `exception`s checked at compile stage.

Reviewer: ```suggestion
- *Checked* : exceptions checked at compile stage.
```

> - *Unchecked*: `exception`s that occur during runtime due to faulty logic. They are direct sub-classes of the `RuntimeException` class.

Reviewer: ```suggestion
- *Unchecked*: exceptions that occur during runtime due to faulty logic. They are direct sub-classes of the `RuntimeException` class.
```

> `Exception`s have to be handled and they can also be thrown. You can handle an `exception` inside a `try/catch` block :  

Reviewer: ```suggestion
Exceptions have to be handled and they can also be thrown. You can handle an exception inside a `try/catch` block :  
```

> An `exception` should be thrown when a fundamental assumption of the current code block is found to be false. 

Reviewer: ```suggestion
An exception should be thrown when a fundamental assumption of the current code block is found to be false. 
```

> For example, a methods that counts the number of elements from a `List` should `throw` an `exception` if the given parameter is not a `List`.

Reviewer: ```suggestion
For example, a methods that counts the number of elements from a `List` should `throw` an exception if the given parameter is not a `List`.
```
